packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -229,12 +229,14 @@ export class Rule extends Resource implements IRule {
 
         // The actual rule lives in the target stack.
         // Other than the account, it's identical to this one
-        new Rule(targetStack, `${this.node.uniqueId}-${id}`, {
+        const rule = new Rule(targetStack, `${this.node.uniqueId}-${id}`, {
           targets: [target],
-          eventPattern: this.eventPattern,
           schedule: this.scheduleExpression ? Schedule.expression(this.scheduleExpression) : undefined,
           description: this.description,
         });
+        // eventPattern is mutable through addEventPattern(), so we need to lazy evaluate it
+        // but only Tokens can be lazy in the framework, so do this magical hack instead
+        (rule as any).eventPattern = this.eventPattern;
```

Comment:
Not entirely sure why this is the way it is. Is it to prevent mutation? Couldn't we leave users to shoot themselves in the foot?

Reply:
This is to make it that both rules (the original, and the new one created in the target's account) have exactly the same `eventPattern`. Because of mutability with `addEventPattern()`, just passing `this.eventPattern` in the constructor of the other rule does not provide that. Hence this brilliant solution/horrible hack (whichever you prefer) to make sure the 2 are identical.